includes/Modules/Analytics_4.php

```diff
@@ -620,6 +615,26 @@ public static function filter_webdatastream_with_ids( $webdatastream ) {
 		return $obj;
 	}
 
+	/**
+	 * Parses a response, adding the _id and _propertyID params and converting to an array keyed by the propertyID and web datastream IDs.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param GoogleAnalyticsAdminV1alphaListWebDataStreamsResponse[] $response Array of GoogleAnalyticsAdminV1alphaListWebDataStreamsResponse objects.
+	 * @return \stdClass[] Array of models containing _id and _propertyID attributes, keyed by the propertyID.
+	 */
+	public static function filter_webdatastreams_batch( $response ) {
+		$mapped         = array();
+		$results        = array_shift( $response );
```

Comment:
There might be more than 1 response in the batch. Thus we need to iterate over each response in the batch, not just the first one.

Reply:
https://github.com/google/site-kit-wp/pull/3755/commits/7886d760d2ff3456eaa4e8dd101be76d379d8702